smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CommandGenerator.java

```diff
@@ -186,6 +194,29 @@ private void writeOutputType(String typeName, Optional<StructureShape> outputSha
         }
     }
 
+    private List<MemberShape> getStramingMembers(StructureShape shape) {
+        return shape.getAllMembers().values().stream().filter(memberShape -> memberShape.hasTrait(StreamingTrait.class))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Ease the input streaming member restriction so that users don't need to construct a stream every time.
+     * This type decoration is allowed in Smithy because it makes input type more permissive than output type
+     * for the same member.
+     * Refer here for more rationales: https://github.com/aws/aws-sdk-js-v3/issues/843
+     */
+    private void writeStreamingInputType(
+            String typeName,
+            Optional<StructureShape> inputShape,
```

Comment:
This should just take the content of the optional, it's already known to exist when this is called and is required by the contents of the function as well.